message_ix/core.py

```diff
@@ -283,6 +266,69 @@ def valid(y_v, y_a):
         v_years, a_years = zip(*year_pairs)
         return pd.DataFrame({'year_vtg': v_years, 'year_act': a_years})
 
+    def firstmodelyear(self):
```

Comment:
Could put `@property` on a line before this, which will make it a read-only attribute of the object.

Reply:
nice! learned something new today!

Reply:
This is actually an API-breaking change. Previously one called `scen.firstmodelyear()`, now you have to do `scen.firstmodelyear` (no parens). If you do, you get the error: 

```
~/.local/lib/python3.7/site-packages/message_ix-1.2.0-py3.7.egg/message_ix/core.py in vintage_and_active_years(self, ya_args, in_horizon)
    245         """
    246         horizon = self.set('year')
--> 247         first = self.firstmodelyear()
    248 
    249         if ya_args:

TypeError: 'int' object is not callable
TypeError: 'int' object is not callable
```

Reply:
I would suggest flagging this and other places where we can refactor to properties and put them in a future release.

Reply:
this is a new feature, so updating the use of `firstmodelyear` as property in the one place where it's used...